pgjdbc/src/main/java/org/postgresql/jdbc/PgResultSet.java

```diff
@@ -1674,16 +1674,20 @@ private void updateRowBuffer() throws SQLException {
       } else {
         switch (getSQLType(columnIndex + 1)) {
 
-          //
-          // toString() isn't enough for date and time types; we must format it correctly
-          // or we won't be able to re-parse it.
-          //
-
+          // boolean needs to be formatted as t or f instead of true or false
+          case Types.BIT:
+          case Types.BOOLEAN:
+            rowBuffer[columnIndex] =
+              connection.encodeString(((Boolean)valueObject).booleanValue() ? "t" : "f");
+            break;
+            //
+            // toString() isn't enough for date and time types; we must format it correctly
+            // or we won't be able to re-parse it.
+            //
           case Types.DATE:
-            rowBuffer[columnIndex] = connection
-                .encodeString(
-                    connection.getTimestampUtils().toString(
-                        getDefaultCalendar(), (Date) valueObject));
+            rowBuffer[columnIndex] =
+              connection.encodeString(connection.getTimestampUtils().toString(getDefaultCalendar(),
+                (Date) valueObject));
```

Comment:
None of the DATE handling changed right? If so let's leave it untouched by the PR.

Reply:
correct. will revert. Didn't like the formatting